components/server/src/workspace/workspace-starter.ts

```diff
@@ -1680,12 +1683,12 @@ export class WorkspaceStarter {
         workspace: Workspace,
         context: WorkspaceContext,
         user: User,
-        lastValidWorkspaceInstanceId: string,
+        fromBackup: boolean,
     ): Promise<{ initializer: WorkspaceInitializer; disposable: Disposable }> {
         let result = new WorkspaceInitializer();
         const disp = new DisposableCollection();
 
-        if (lastValidWorkspaceInstanceId != "") {
+        if (fromBackup) {
```

Comment:
This check was the only reason we piped `lastValidWorkspaceInstanceId: string` through 4 layers... :see_no_evil: